public/app/features/search/page/data.ts

```diff
@@ -0,0 +1,114 @@
+import { ArrayVector, DataFrame, Field, FieldType } from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { GrafanaDatasource } from 'app/plugins/datasource/grafana/datasource';
+import { GrafanaQueryType } from 'app/plugins/datasource/grafana/types';
+import { lastValueFrom } from 'rxjs';
+
+export interface DashboardData {
+  dashboards: DataFrame;
+  panels: DataFrame;
+  panelTypes: DataFrame;
+  schemaVersions: DataFrame;
+}
+
+export async function getDashboardData(): Promise<DashboardData> {
```

Comment:
Probably would be a good idea to provide some unit tests for these helper functions